nix/services/default.nix

```diff
@@ -24,4 +25,30 @@ in
     ./searxng.nix
     ./tika.nix
   ];
+
+  options.services.outputs.devShell = lib.mkOption {
+    type = lib.types.package;
+    readOnly = true;
+    description = ''
+      The devShell that aggregates packages from all the enabled services.
+    '';
+  };
+
+  config = {
+    services.outputs.devShell = pkgs.mkShell {
+      packages =
+        lib.pipe config.services [
+          # `outputs` is a reserved attribute set and is not the name of a service.
+          (lib.filterAttrs (n: _: n != "outputs"))
+          # Flatten services attrset
+          #
+          # Example:
+          # Input = { mysql."m1" = <cfg1>; mysql."m2" = <cfg2>; redis."r1" = <cfg3>; }
+          # Output = [ <cfg1> <cfg2> <cfg3> ]
+          (lib.foldlAttrs (acc: _: v: (lib.attrValues v) ++ acc) [ ])
+          (lib.filter (service: service.enable))
+          (map (service: service.package))
```

Comment:
This assumes that all services -- and services to be added in future -- have `package` option, right?

Reply:
Yes

Reply:
An error is thrown if a service doesn’t define `package` option, since https://github.com/juspay/services-flake/pull/355/commits/72abbb7d91c27ec3a7602dbda262570c46b8be8e. 

We can safely assume every service will have a package, if there is an exception, we can revisit this in the future.